### PR TITLE
[jsk_perception] squeeze mask to image dim=2 in fcn segmentation

### PR DESCRIPTION
**Pull request by knorth55**
- modify `mask.shape` from `(width, height, channel=1)` to `(width, height)`, which is same as `label.shape`
- fix typo

### CODE REVIEW

**Comment by wkentaro** [comment]
could you please add sample and test?
You can use mask_image_creator.

**Comment by knorth55** [comment]
I added sample, test and image in doc.

File: jsk_perception/test/fcn_object_segmentation_with_mask.test [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <arg name="gui" value="false" />
+    <arg name="use_mask" value="true" />
+  </include>
+
+  <test test-name="test_fcn_object_segmentation"
+        name="test_fcn_object_segmentation"
+        pkg="jsk_tools" type="test_topic_published.py"
+        retry="3">
+    <rosparam>
+      topic_0: /fcn_object_segmentation/output
```
Comment (wkentaro) [inline jsk_perception/test/fcn_object_segmentation_with_mask.test]: why not add topic_1 to fcn_object_segmentation.test?

Comment (knorth55) [inline jsk_perception/test/fcn_object_segmentation_with_mask.test]: ok, make two test file to one.

**Comment by k-okada** [comment]
humm, could not pass https://travis-ci.org/jsk-ros-pkg/jsk_recognition/jobs/250268417
```


Traceback (most recent call last):




  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback




    cb(msg)




  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 74, in callback




    self.signalMessage(msg)




  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 56, in signalMessage




    cb(*(msg + args))




  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 199, in add




    self.signalMessage(*msgs)




  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 56, in signalMessage




    cb(*(msg + args))




  File "/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/node_scripts/fcn_object_segmentation.py", line 138, in _cb_with_mask




    mask = np.squeeze(mask, axis=2)




  File "/usr/local/lib/python2.7/dist-packages/numpy/core/fromnumeric.py", line 1198, in squeeze




    return squeeze(axis=axis)




AxisError: axis 2 is out of bounds for array of dimension 2











testtest_fcn_object_segmentation ... ERROR!




ERROR: max time [60.0s] allotted for test [test_fcn_object_segmentation] of type [jsk_tools/test_topic_published.py]





  File "/usr/lib/python2.7/unittest/case.py", line 331, in run




    testMethod()




  File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/runner.py", line 148, in fn




    self.test_parent.run_test(test)




  File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/rostest_parent.py", line 132, in run_test




    return self.runner.run_test(test)




  File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/launch.py", line 681, in run_test




    (test.time_limit, test.test_name, test.package, test.type))

```

**Comment by k-okada** [comment]
```
Finished  <<< jsk_perception                         [ 10 minutes and 54.0 seconds ]
[build] Summary: All 5 packages succeeded!
[build]   Ignored:   10 packages were skipped or are blacklisted.
[build]   Warnings:  1 packages succeeded with warnings.
[build]   Abandoned: None.
[build]   Failed:    None.
[build] Runtime: 11 minutes and 25.5 seconds total.
+ catkin_test_results --verbose --all build
Skipping "catkin_tools_prebuild/package.xml": 'tests'
Skipping "jsk_perception/test_results/jsk_perception/rosunit-test_people_pose_estimation_2d.xml": no element found: line 2, column 0
Full test results for 'jsk_perception/test_results/jsk_perception/rostest-test_fcn_object_segmentation.xml'
-------------------------------------------------
<?xml version="1.0" encoding="utf-8"?>
<testsuite errors="1" failures="0" name="unittest.suite.TestSuite" tests="1" time="249.925">
  <testcase classname="rostest.runner.RosTest" name="testtest_fcn_object_segmentation" time="249.9246">
    <error type="RLTestTimeoutException">max time [60.0s] allotted for test [test_fcn_object_segmentation] of type [jsk_tools/test_topic_published.py]
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/runner.py", line 148, in fn
    self.test_parent.run_test(test)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/rostest_parent.py", line 132, in run_test
    return self.runner.run_test(test)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/launch.py", line 681, in run_test
    (test.time_limit, test.test_name, test.package, test.type))
    </error>
  </testcase>
  <system-out><![CDATA[[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[cb84ceda-61d9-11e7-b491-0242ac110002] starting
?[1mstarted roslaunch server http://92d30dc908ca:50342/?[0m

SUMMARY
========

PARAMETERS
 * /no_mask/fcn_object_segmentation/gpu: -1
 * /no_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /no_mask/fcn_object_segmentation/model_name: fcn8s
 * /no_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /no_mask/fcn_object_segmentation/use_mask: False
 * /no_mask/image_publisher/file_name: /workspace/ros/ws...
 * /no_mask/image_publisher/publish_info: False
 * /no_mask/label_image_decomposer/queue_size: 100
 * /no_mask/mask_image_generator/height: 420
 * /no_mask/mask_image_generator/offset_x: 110
 * /no_mask/mask_image_generator/offset_y: 25
 * /no_mask/mask_image_generator/width: 280
 * /rosdistro: indigo
 * /rosversion: 1.11.21
 * /test_fcn_object_segmentation/timeout_0: 60
 * /test_fcn_object_segmentation/timeout_1: 60
 * /test_fcn_object_segmentation/topic_0: /no_mask/fcn_obje...
 * /test_fcn_object_segmentation/topic_1: /use_mask/fcn_obj...
 * /use_mask/fcn_object_segmentation/gpu: -1
 * /use_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /use_mask/fcn_object_segmentation/model_name: fcn8s
 * /use_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /use_mask/fcn_object_segmentation/use_mask: True
 * /use_mask/image_publisher/file_name: /workspace/ros/ws...
 * /use_mask/image_publisher/publish_info: False
 * /use_mask/label_image_decomposer/queue_size: 100
 * /use_mask/mask_image_generator/height: 420
 * /use_mask/mask_image_generator/offset_x: 110
 * /use_mask/mask_image_generator/offset_y: 25
 * /use_mask/mask_image_generator/width: 280

NODES
  /use_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)
  /no_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)

[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[cb84ceda-61d9-11e7-b491-0242ac110002] done
[ROSTEST]Running test [test_fcn_object_segmentation]
?[1mROS_MASTER_URI=http://92d30dc908ca:60853/?[0m
?[1msetting /run_id to cb84ceda-61d9-11e7-b491-0242ac110002?[0m
?[1mprocess[rosout-1]: started with pid [36973]?[0m
started core service [/rosout]
?[1mprocess[no_mask/image_publisher-2]: started with pid [36990]?[0m
?[1mprocess[no_mask/mask_image_generator-3]: started with pid [36991]?[0m
?[1mprocess[no_mask/fcn_object_segmentation-4]: started with pid [37031]?[0m
?[1mprocess[no_mask/label_image_decomposer-5]: started with pid [37032]?[0m
?[1mprocess[use_mask/image_publisher-6]: started with pid [37090]?[0m
?[1mprocess[use_mask/mask_image_generator-7]: started with pid [37103]?[0m
?[1mprocess[use_mask/fcn_object_segmentation-8]: started with pid [37153]?[0m
?[1mprocess[use_mask/label_image_decomposer-9]: started with pid [37171]?[0m
[ROSTEST]running test test_fcn_object_segmentation
?[1mprocess[test_fcn_object_segmentation-10]: started with pid [37260]?[0m
[ROSTEST]test [test_fcn_object_segmentation] failed, retrying. Retries left: 2
[ROSTEST]tearDown[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test]
shutting down processing monitor...
[test_fcn_object_segmentation-10] killing on exit
[use_mask/label_image_decomposer-9] killing on exit
[use_mask/mask_image_generator-7] killing on exit
[use_mask/fcn_object_segmentation-8] killing on exit
[no_mask/image_publisher-2] killing on exit
[use_mask/image_publisher-6] killing on exit
[no_mask/label_image_decomposer-5] killing on exit
[no_mask/fcn_object_segmentation-4] killing on exit
[no_mask/mask_image_generator-3] killing on exit
[rosout-1] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test complete
[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[f0b5006c-61d9-11e7-b491-0242ac110002] starting
?[1mstarted roslaunch server http://92d30dc908ca:44585/?[0m

SUMMARY
========

PARAMETERS
 * /no_mask/fcn_object_segmentation/gpu: -1
 * /no_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /no_mask/fcn_object_segmentation/model_name: fcn8s
 * /no_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /no_mask/fcn_object_segmentation/use_mask: False
 * /no_mask/image_publisher/file_name: /workspace/ros/ws...
 * /no_mask/image_publisher/publish_info: False
 * /no_mask/label_image_decomposer/queue_size: 100
 * /no_mask/mask_image_generator/height: 420
 * /no_mask/mask_image_generator/offset_x: 110
 * /no_mask/mask_image_generator/offset_y: 25
 * /no_mask/mask_image_generator/width: 280
 * /rosdistro: indigo
 * /rosversion: 1.11.21
 * /test_fcn_object_segmentation/timeout_0: 60
 * /test_fcn_object_segmentation/timeout_1: 60
 * /test_fcn_object_segmentation/topic_0: /no_mask/fcn_obje...
 * /test_fcn_object_segmentation/topic_1: /use_mask/fcn_obj...
 * /use_mask/fcn_object_segmentation/gpu: -1
 * /use_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /use_mask/fcn_object_segmentation/model_name: fcn8s
 * /use_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /use_mask/fcn_object_segmentation/use_mask: True
 * /use_mask/image_publisher/file_name: /workspace/ros/ws...
 * /use_mask/image_publisher/publish_info: False
 * /use_mask/label_image_decomposer/queue_size: 100
 * /use_mask/mask_image_generator/height: 420
 * /use_mask/mask_image_generator/offset_x: 110
 * /use_mask/mask_image_generator/offset_y: 25
 * /use_mask/mask_image_generator/width: 280

NODES
  /use_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)
  /no_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)

[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[f0b5006c-61d9-11e7-b491-0242ac110002] done
[ROSTEST]Running test [test_fcn_object_segmentation]
?[1mROS_MASTER_URI=http://92d30dc908ca:49193/?[0m
?[1msetting /run_id to f0b5006c-61d9-11e7-b491-0242ac110002?[0m
?[1mprocess[rosout-11]: started with pid [37774]?[0m
started core service [/rosout]
?[1mprocess[no_mask/image_publisher-12]: started with pid [37791]?[0m
?[1mprocess[no_mask/mask_image_generator-13]: started with pid [37792]?[0m
?[1mprocess[no_mask/fcn_object_segmentation-14]: started with pid [37832]?[0m
?[1mprocess[no_mask/label_image_decomposer-15]: started with pid [37833]?[0m
?[1mprocess[use_mask/image_publisher-16]: started with pid [37891]?[0m
?[1mprocess[use_mask/mask_image_generator-17]: started with pid [37931]?[0m
?[1mprocess[use_mask/fcn_object_segmentation-18]: started with pid [37932]?[0m
?[1mprocess[use_mask/label_image_decomposer-19]: started with pid [37972]?[0m
[ROSTEST]removing previous test results file [/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/rosunit-test_fcn_object_segmentation.xml]
[ROSTEST]running test test_fcn_object_segmentation
?[1mprocess[test_fcn_object_segmentation-20]: started with pid [38004]?[0m
[ROSTEST]test [test_fcn_object_segmentation] failed, retrying. Retries left: 1
[ROSTEST]tearDown[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test]
shutting down processing monitor...
[test_fcn_object_segmentation-20] killing on exit
[use_mask/label_image_decomposer-19] killing on exit
[use_mask/fcn_object_segmentation-18] killing on exit
[use_mask/image_publisher-16] killing on exit
[no_mask/image_publisher-12] killing on exit
[no_mask/label_image_decomposer-15] killing on exit
[no_mask/fcn_object_segmentation-14] killing on exit
[no_mask/mask_image_generator-13] killing on exit
[use_mask/mask_image_generator-17] killing on exit
[rosout-11] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test complete
[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[16077a0c-61da-11e7-b491-0242ac110002] starting
?[1mstarted roslaunch server http://92d30dc908ca:55570/?[0m

SUMMARY
========

PARAMETERS
 * /no_mask/fcn_object_segmentation/gpu: -1
 * /no_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /no_mask/fcn_object_segmentation/model_name: fcn8s
 * /no_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /no_mask/fcn_object_segmentation/use_mask: False
 * /no_mask/image_publisher/file_name: /workspace/ros/ws...
 * /no_mask/image_publisher/publish_info: False
 * /no_mask/label_image_decomposer/queue_size: 100
 * /no_mask/mask_image_generator/height: 420
 * /no_mask/mask_image_generator/offset_x: 110
 * /no_mask/mask_image_generator/offset_y: 25
 * /no_mask/mask_image_generator/width: 280
 * /rosdistro: indigo
 * /rosversion: 1.11.21
 * /test_fcn_object_segmentation/timeout_0: 60
 * /test_fcn_object_segmentation/timeout_1: 60
 * /test_fcn_object_segmentation/topic_0: /no_mask/fcn_obje...
 * /test_fcn_object_segmentation/topic_1: /use_mask/fcn_obj...
 * /use_mask/fcn_object_segmentation/gpu: -1
 * /use_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /use_mask/fcn_object_segmentation/model_name: fcn8s
 * /use_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /use_mask/fcn_object_segmentation/use_mask: True
 * /use_mask/image_publisher/file_name: /workspace/ros/ws...
 * /use_mask/image_publisher/publish_info: False
 * /use_mask/label_image_decomposer/queue_size: 100
 * /use_mask/mask_image_generator/height: 420
 * /use_mask/mask_image_generator/offset_x: 110
 * /use_mask/mask_image_generator/offset_y: 25
 * /use_mask/mask_image_generator/width: 280

NODES
  /use_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)
  /no_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)

[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[16077a0c-61da-11e7-b491-0242ac110002] done
[ROSTEST]Running test [test_fcn_object_segmentation]
?[1mROS_MASTER_URI=http://92d30dc908ca:36316/?[0m
?[1msetting /run_id to 16077a0c-61da-11e7-b491-0242ac110002?[0m
?[1mprocess[rosout-21]: started with pid [38575]?[0m
started core service [/rosout]
?[1mprocess[no_mask/image_publisher-22]: started with pid [38592]?[0m
?[1mprocess[no_mask/mask_image_generator-23]: started with pid [38593]?[0m
?[1mprocess[no_mask/fcn_object_segmentation-24]: started with pid [38594]?[0m
?[1mprocess[no_mask/label_image_decomposer-25]: started with pid [38634]?[0m
?[1mprocess[use_mask/image_publisher-26]: started with pid [38635]?[0m
?[1mprocess[use_mask/mask_image_generator-27]: started with pid [38693]?[0m
?[1mprocess[use_mask/fcn_object_segmentation-28]: started with pid [38733]?[0m
?[1mprocess[use_mask/label_image_decomposer-29]: started with pid [38773]?[0m
[ROSTEST]removing previous test results file [/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/rosunit-test_fcn_object_segmentation.xml]
[ROSTEST]running test test_fcn_object_segmentation
?[1mprocess[test_fcn_object_segmentation-30]: started with pid [38774]?[0m
[ROSTEST]test [test_fcn_object_segmentation] failed, retrying. Retries left: 0
[ROSTEST]tearDown[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test]
shutting down processing monitor...
[test_fcn_object_segmentation-30] killing on exit
[use_mask/label_image_decomposer-29] killing on exit
[use_mask/mask_image_generator-27] killing on exit
[use_mask/fcn_object_segmentation-28] killing on exit
[no_mask/fcn_object_segmentation-24] killing on exit
[use_mask/image_publisher-26] killing on exit
[no_mask/label_image_decomposer-25] killing on exit
[no_mask/image_publisher-22] killing on exit
[no_mask/mask_image_generator-23] killing on exit
[rosout-21] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test complete
[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[3b771db0-61da-11e7-b491-0242ac110002] starting
?[1mstarted roslaunch server http://92d30dc908ca:52020/?[0m

SUMMARY
========

PARAMETERS
 * /no_mask/fcn_object_segmentation/gpu: -1
 * /no_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /no_mask/fcn_object_segmentation/model_name: fcn8s
 * /no_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /no_mask/fcn_object_segmentation/use_mask: False
 * /no_mask/image_publisher/file_name: /workspace/ros/ws...
 * /no_mask/image_publisher/publish_info: False
 * /no_mask/label_image_decomposer/queue_size: 100
 * /no_mask/mask_image_generator/height: 420
 * /no_mask/mask_image_generator/offset_x: 110
 * /no_mask/mask_image_generator/offset_y: 25
 * /no_mask/mask_image_generator/width: 280
 * /rosdistro: indigo
 * /rosversion: 1.11.21
 * /test_fcn_object_segmentation/timeout_0: 60
 * /test_fcn_object_segmentation/timeout_1: 60
 * /test_fcn_object_segmentation/topic_0: /no_mask/fcn_obje...
 * /test_fcn_object_segmentation/topic_1: /use_mask/fcn_obj...
 * /use_mask/fcn_object_segmentation/gpu: -1
 * /use_mask/fcn_object_segmentation/model_file: /workspace/ros/ws...
 * /use_mask/fcn_object_segmentation/model_name: fcn8s
 * /use_mask/fcn_object_segmentation/target_names: ['background', 'a...
 * /use_mask/fcn_object_segmentation/use_mask: True
 * /use_mask/image_publisher/file_name: /workspace/ros/ws...
 * /use_mask/image_publisher/publish_info: False
 * /use_mask/label_image_decomposer/queue_size: 100
 * /use_mask/mask_image_generator/height: 420
 * /use_mask/mask_image_generator/offset_x: 110
 * /use_mask/mask_image_generator/offset_y: 25
 * /use_mask/mask_image_generator/width: 280

NODES
  /use_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)
  /no_mask/
    fcn_object_segmentation (jsk_perception/fcn_object_segmentation.py)
    image_publisher (jsk_perception/image_publisher.py)
    label_image_decomposer (jsk_perception/label_image_decomposer.py)
    mask_image_generator (jsk_perception/mask_image_generator)

[ROSTEST]setup[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test] run_id[3b771db0-61da-11e7-b491-0242ac110002] done
[ROSTEST]Running test [test_fcn_object_segmentation]
?[1mROS_MASTER_URI=http://92d30dc908ca:50777/?[0m
?[1msetting /run_id to 3b771db0-61da-11e7-b491-0242ac110002?[0m
?[1mprocess[rosout-31]: started with pid [39375]?[0m
started core service [/rosout]
?[1mprocess[no_mask/image_publisher-32]: started with pid [39392]?[0m
?[1mprocess[no_mask/mask_image_generator-33]: started with pid [39393]?[0m
?[1mprocess[no_mask/fcn_object_segmentation-34]: started with pid [39394]?[0m
?[1mprocess[no_mask/label_image_decomposer-35]: started with pid [39434]?[0m
?[1mprocess[use_mask/image_publisher-36]: started with pid [39435]?[0m
?[1mprocess[use_mask/mask_image_generator-37]: started with pid [39501]?[0m
?[1mprocess[use_mask/fcn_object_segmentation-38]: started with pid [39572]?[0m
?[1mprocess[use_mask/label_image_decomposer-39]: started with pid [39573]?[0m
[ROSTEST]removing previous test results file [/workspace/ros/ws_jsk_recognition/build/jsk_perception/test_results/jsk_perception/rosunit-test_fcn_object_segmentation.xml]
[ROSTEST]running test test_fcn_object_segmentation
?[1mprocess[test_fcn_object_segmentation-40]: started with pid [39574]?[0m
[ROSTEST]tearDown[/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test]
shutting down processing monitor...
[test_fcn_object_segmentation-40] killing on exit
[use_mask/label_image_decomposer-39] killing on exit
[use_mask/fcn_object_segmentation-38] killing on exit
[use_mask/mask_image_generator-37] killing on exit
[use_mask/image_publisher-36] killing on exit
[no_mask/label_image_decomposer-35] killing on exit
[no_mask/fcn_object_segmentation-34] killing on exit
[no_mask/image_publisher-32] killing on exit
[no_mask/mask_image_generator-33] killing on exit
[rosout-31] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown /workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_perception/test/fcn_object_segmentation.test complete
]]></system-out>
  <system-err><![CDATA[[ROSTEST]test [test_fcn_object_segmentation] timed out
[ROSTEST]test [test_fcn_object_segmentation] timed out
[ROSTEST]test [test_fcn_object_segmentation] timed out
]]></system-err>
</testsuite>
```

**Comment by knorth55** [comment]
it seems that `cv_bridge` returns `ndim=2` array with `mono8` when we use `opencv3`.
`cv_bridge` returns `ndim=3` array when we use `opencv2` instead.

**Comment by knorth55** [comment]
I solved the problem.
Could you please merge this?

**Comment by knorth55** [comment]
@k-okada could you please merge this?

**Comment by knorth55** [comment]
> it seems that cv_bridge returns ndim=2 array with mono8 when we use opencv3.
cv_bridge returns ndim=3 array when we use opencv2 instead.



I made an issue.
https://github.com/ros-perception/vision_opencv/issues/177

**Pull request merged**